ctx.go

```diff
@@ -365,6 +365,55 @@ func (c *netHttpContext[B, P]) Body() (B, error) {
 	return body, err
 }
 
+func bitSize(kind reflect.Kind) int {
+	switch kind {
+	case reflect.Uint8, reflect.Int8:
+		return 8
+	case reflect.Uint16, reflect.Int16:
+		return 16
+	case reflect.Uint32, reflect.Int32, reflect.Float32:
+		return 32
+	case reflect.Uint, reflect.Int:
+		return strconv.IntSize
+	}
+	return 64
+}
+
+// setParamValue sets a value to a reflect.Value based on its kind
+func setParamValue(value reflect.Value, paramValue string, kind reflect.Kind) error {
+	switch kind {
+	case reflect.String:
+		value.SetString(paramValue)
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		intValue, err := strconv.ParseInt(paramValue, 10, bitSize(kind))
+		if err != nil {
+			return fmt.Errorf("cannot convert %s to int: %w", paramValue, err)
+		}
+		value.SetInt(intValue)
```

Comment:
Just found this

https://pkg.go.dev/reflect#Value.OverflowInt

I would like to suggest this

```suggestion
	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
		intValue, err := strconv.ParseInt(paramValue, 10, 64)
		if err != nil {
			return fmt.Errorf("cannot convert %s to %s: %w", paramValue, kind, err)
		}
		if value.OverflowInt(intValue) {
			return fmt.Errorf("%s overflows %s", paramValue, kind)
		}
		value.SetInt(intValue)
```

I didn't test

Reply:
Interesting!